.github/workflows/ci.yaml

```diff
@@ -0,0 +1,108 @@
+name: Continuous Integration
+
+on:
+  pull_request:
+  release:
+    types: [published]
+
+env:
+  PROJECT_NAME: ml-recipe-midi-generator
+  NEURO_STAGING_URL: ${{ secrets.NEURO_STAGING_URL }}
+  NEURO_TOKEN: ${{ secrets.NEURO_TOKEN }}
+
+jobs:
+  test:
+    name: Build and test image
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout commit
+        uses: actions/checkout@v2
+      - name: Setup Environment Variables
+        run: |
+          echo ::set-env name=PROJECT_PATH_STORAGE::storage:build-$PROJECT_NAME-$GITHUB_RUN_ID
+          echo ::set-env name=IMAGE_NAME::neuromation/$PROJECT_NAME
+          echo ::set-env name=IMAGE_TAG::build-$GITHUB_RUN_ID
+          echo ::set-env name=JOBS_TAG::build-$GITHUB_RUN_ID
+      - name: Install python
+        uses: actions/setup-python@v2
+        with:
+          python-version: "3.7"
+      - name: Install dependencies
+        run: |
+          python -m venv venv
+          source venv/bin/activate
+
+          pip install -r .github/workflows/requirements.txt
+      - name: Configure environment
+        run: |
+          source venv/bin/activate
+
+          neuro config login-with-token $NEURO_TOKEN $NEURO_STAGING_URL
+          neuro config show
+      - name: Build and test image
+        run: |
+          source venv/bin/activate
+
+          export CUSTOM_ENV_NAME=image:$IMAGE_NAME:$IMAGE_TAG
+          make setup __BAKE_SETUP=yes | tee
+          export RUN_EXTRA="--tag=$JOBS_TAG"
+
+          make -f test.mk test_jupyter TRAINING_MACHINE_TYPE=cpu-small | tee
+          make -f test.mk test_jupyter TRAINING_MACHINE_TYPE=gpu-small | tee
+          make -f test.mk test_jupyter_baked TRAINING_MACHINE_TYPE=cpu-small | tee
+          make -f test.mk test_jupyter_baked TRAINING_MACHINE_TYPE=gpu-small | tee
+      - name: Cleanup Jobs
+        if: ${{ always() }}
+        run: |
+          source venv/bin/activate
+
+          export JOBS_LEFT=$(neuro -q ps --tag=$JOBS_TAG)
+          [ ! "$JOBS_LEFT" ] || neuro kill $JOBS_LEFT
+  deploy:
+    name: Run tests
```

Comment:
Minor: Better name: `Build, test and publish image` or just `Deploy image`

Reply:
Agree